### PR TITLE
refactor(manager): add check for operations when commit id is None

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

File: graviti/manager/branch.py [hunk 1]
```diff
@@ -79,12 +79,19 @@ def create(self, name: str, revision: Optional[str] = None) -> Branch:
...
+
         Returns:
             The :class:`.Branch` instance with the given name.
 
         """
         if not revision:
             revision = self._dataset.HEAD.commit_id
+            if revision is None:
+                raise ForbiddenError(
+                    "Can not create tag on the default branch without commit history"
```
Comment (linjiX) [inline graviti/manager/branch.py:93]: I think the word `Can not` is not clear enough.
Will it be better if we use `xxx is not allowed` or `xxx is forbidden` to point out that this behavior is forbidden by design not other reason?

File: graviti/manager/commit.py [hunk 2]
```diff
@@ -84,6 +85,11 @@ def _list_data(self, offset: int, limit: int, sheet_name: str) -> Dict[str, Any]
         )["data"]
 
     def _list_sheets(self) -> Dict[str, Any]:
+        if self.commit_id is None:
+            raise StatusError(
+                "No commit on the current branch. Please create a draft and add data first"
```
Comment (linjiX) [inline graviti/manager/commit.py:90]: why we need add data? can the user just create an empty draft?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/commit.py [hunk 3]
```diff
@@ -267,12 +280,17 @@ def get(self, revision: Optional[str] = None) -> Commit:
             revision: The information to locate the specific commit, which can be the commit id,
```
Comment (graczhual) [inline graviti/manager/commit.py:280]: do we need to add if-else to `_generate commits`

Comment (Lee-000) [reply graviti/manager/commit.py:280]: If the commit id is None, the revision won't be added to the params of the request.

**Review by linjiX — APPROVED** [review]

**Review by graczhual — APPROVED** [review]

**Pull request merged**